### ISSUE TITLE
Does not automatically adds adobe_xd plugin to pubspec.yaml

### ISSUE THREAD

**Issue by hemish11** [AdobeXD/xd-to-flutter-plugin#26]
This is a feature request and not a bug.
While exporting all the widgets in adobe xd or adding project to it, it does not automatically adds adobe_xd plugin. It would be much easy for user so that user would run application easily without adding the plugin. If it was so, then user can start application suddenly after exporting all the widgets and would only need to import them in main.dart file. Hence it would be much easy for them to use the plugin and integrate them in their flutter app.
To do this, the plugin can edit pubspec.yaml file and then can run `flutter pub get` in terminal by itself and hence user would not need to do anything. This same could be done for adding assets entry and adding fonts entry. 
For beginner they thus don't need to worry to edit pubspec.yaml. As properly indenting pubspec .yaml is very important and it is much hard for Beginner. Thus it would be much helpful for them also and can easily create amazing ui with the power of this plugin and flutter.

**Comment by gskinner** [comment]
We don't currently do this for two reasons:
1. We are starting with a cautious philosophy of not touching critical files in your project, to ensure that we don't break anything too badly. This could change with future versions, but we wanted to play it safe initially.
2. Plugins in XD have extremely limited system access (good for user security, bad for robust plugins), so we can't run terminal commands. This means that even if we write the dependency, you may still need to run the package update.

Those two things combined (plus time) meant we didn't want to tackle this for the initial release. Keeping this one open for future implementation though. This also applies to fonts and image paths.

We could mitigate some of the concerns by asking the user if they want us to do this before we do - this would require some UI work too.

**Comment by gskinner** [comment]
I'm actually going to keep this open if that's ok, and just give it a bit broader name.

**Comment by cyrsis** [comment]
Yes, XD is bad ass, I always have to  write some extra dart code to work with their assets export

**Comment by gskinner** [comment]
Going to look at this together with #80 for v2

**Comment by hemish11** [comment]
We can also do it like, that on button press, xd would copy the whole pubspec.yaml file in the clipboard and then we would paste it in our file. So no need to edit any file.

**Comment by connelevalsam** [comment]
Hello, I'm new to flutter, I did a design in XD and exported to Flutter then copied the files to my project but have some errors

**Comment by hemish11** [comment]
Are there any assets in the project line image or font or so?

**Comment by hemish11** [comment]
I am really sorry for the late reply, as there was no notification from GitHub to me. So my question is that are there any assets in the project, Like images, fonts, etc. If so, you need to mention those in pubspec.yaml. And for all project, you need to add some packages and run flutter pub get

**Comment by connelevalsam** [comment]
Thanks for the reply.

Yeah I have all that, my major problem is navigation or button clicks.

Even when I tried to add InkWell() it didn't work like it's so frustrating and why did Adobe use transform.translate when Flutter uses InkWell or Gesture

**Comment by gskinner** [comment]
I made significant improvements to how we read the pubspec.yaml file in v2.0.0, and added more robust checks (ex. checking the Dart SDK version for null safety). Writing back to the pubspec file is still something I'm a bit hesitant about because I don't want to overwrite comments and such, but I have found an NPM package that may allow this in the future:
https://www.npmjs.com/package/yaml
